cocotb/triggers.py

```diff
@@ -655,18 +656,24 @@ class _AggregateWaitable(Waitable):
     __slots__ = ('triggers',)
 
     def __init__(self, *triggers):
-        self.triggers = tuple(triggers)
+        usable_triggers = []
 
         # Do some basic type-checking up front, rather than waiting until we
         # await them.
         allowed_types = (Trigger, Waitable, cocotb.decorators.RunningTask)
-        for trigger in self.triggers:
-            if not isinstance(trigger, allowed_types):
+        for trigger in triggers:
+            if inspect.iscoroutine(trigger):
+                usable_triggers.append(cocotb.decorators.RunningTask(trigger))
+            elif isinstance(trigger, allowed_types):
+                usable_triggers.append(trigger)
+            else:
```

Comment:
Do we want to explicitly add coroutine objects to the list of acceptable types or change the predicate to `inspect.isawaitable` since the waiter coroutine just `await`s on each object?

Reply:
I thought about that, but `isawaitable` supports undecorated generators, which we don't support anywhere else that I know of.

Reply:
> but isawaitable supports undecorated generators

You mean `asyncio.coroutine` or `types.coroutine` decorated generators? Regular generators are not awaitable. However, they would likely work (if not so rigidly type-checked by the scheduler) since they are coroutine-like, but I guess `isawaitable` is not the right way to do that.